File: src/main/kotlin/unizar/labis/g03/backendapp/model/DTO/ReservaDTO.java
package unizar.labis.g03.backendapp.model.DTO;

import lombok.Getter;
import unizar.labis.g03.backendapp.model.valueObjects.InfoReserva;

import java.time.LocalDateTime;
import java.util.List;

public class ReservaDTO {
    @Getter
    private final List<String> idEspacios;
    @Getter
    private final String emailUsuario;
    private final LocalDateTime fechaInicio;
    private final LocalDateTime fechaFinal;
    private final int numAsistentesPrevistos;
    private final String descripcion;

    // Constructor
    public ReservaDTO(List<String> idEspacios, String emailUsuario, LocalDateTime fechaInicio, LocalDateTime fechaFinal,
                      int numAsistentesPrevistos, String descripcion) {
        this.idEspacios = idEspacios;
        this.emailUsuario = emailUsuario;
        this.fechaInicio = fechaInicio;
        this.fechaFinal = fechaFinal;
        this.numAsistentesPrevistos = numAsistentesPrevistos;
        this.descripcion = descripcion;
    }
    public InfoReserva getInfoReserva() {
        return new InfoReserva(numAsistentesPrevistos,fechaInicio, fechaFinal, descripcion,0);
    }

}

File: src/main/kotlin/unizar/labis/g03/backendapp/model/DTO/ReservaDTO.kt
package unizar.labis.g03.backendapp.model.DTO

import lombok.Getter
import unizar.labis.g03.backendapp.model.valueObjects.InfoReserva
import java.time.LocalDateTime

class ReservaDTO // Constructor
    (
    @field:Getter private val idEspacios: List<String>,
    @field:Getter private val emailUsuario: String,
    private val fechaInicio: LocalDateTime,
    private val fechaFinal: LocalDateTime,
    private val numAsistentesPrevistos: Int,
    private val descripcion: String
) {
    fun getIdEspacios(): List<String> {
        return idEspacios
    }
    fun getEmailUsuario(): String {
        return emailUsuario
    }
    fun getFechaInicio(): LocalDateTime {
        return fechaInicio
    }
    fun getFechaFinal(): LocalDateTime {
        return fechaFinal
    }
    fun getNumAsistentesPrevistos(): Int {
        return numAsistentesPrevistos
    }

    fun getDescripcion(): String {
        return descripcion
    }
}
